commands/createfixpullrequests.go

```diff
@@ -307,42 +310,54 @@ func (cfp *CreateFixPullRequestsCmd) updatePackageToFixedVersion(packageType cor
 	}
 
 	switch packageType {
-	case coreutils.Go:
-		commandArgs := []string{"get"}
-		err = fixPackageVersionGeneric(packageType.GetExecCommandName(), commandArgs, impactedPackage, fixVersion, "@v")
-	case coreutils.Npm:
-		commandArgs := []string{"install"}
-		err = fixPackageVersionGeneric(packageType.GetExecCommandName(), commandArgs, impactedPackage, fixVersion, "@")
 	case coreutils.Maven:
 		err = fixPackageVersionMaven(cfp, impactedPackage, fixVersion)
-	case coreutils.Yarn:
-		commandArgs := []string{"up"}
-		err = fixPackageVersionGeneric(packageType.GetExecCommandName(), commandArgs, impactedPackage, fixVersion, "@")
 	case coreutils.Pip:
 		err = fixPackageVersionPip(impactedPackage, fixVersion, requirementsFile)
-	case coreutils.Pipenv:
-		commandArgs := []string{"install"}
-		err = fixPackageVersionGeneric(packageType.GetExecCommandName(), commandArgs, impactedPackage, fixVersion, "==")
 	case coreutils.Poetry:
 		err = fixPackageVersionPoetry(impactedPackage, fixVersion)
 	default:
-		return fmt.Errorf("package type: %s is currently not supported", string(packageType))
+		err = fixPackageVersionGeneric(packageType, impactedPackage, fixVersion)
+		if err != nil {
+			return err
+		}
 	}
-
 	return
 }
 
 // The majority of package managers already support upgrading specific package versions and update the dependency files automatically.
 // In other cases, we had to handle the upgrade process
-// commandName - Name of the package manager
-// commandArgs - Package manager upgrade command
+// technology - Name of the package manager
 // impactedPackage - Vulnerable package to upgrade
 // fixVersion - The version that fixes the vulnerable package
-// operator - The operator between the impactedPackage to the fixVersion
-func fixPackageVersionGeneric(commandName string, commandArgs []string, impactedPackage, fixVersion, operator string) error {
+func fixPackageVersionGeneric(technology coreutils.Technology, impactedPackage, fixVersion string) error {
+
+	if technology == coreutils.Go {
```

Comment:
Wouldn't it be better tp place this code inside a new `fixPackageVersionGo` to remain consistent with tthe way the code is currently structured? We have `fixPackageVersionMaven` and a few other similar functions. Why should we break the code symetry?

Reply:
I've looked and there is only one other use to this function. (in the poetry function)
wouldn't it be more generic if you didn't have to pass in the operator and the command name? 
instead you pass in the tech which has all of the constants.